File: LeetCode/Python/ComplementOfBase10Integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        const=0
        ans=0
        if n==0:
            return 1
        while n>0:
            if n&1:
                const+=1
            else:
                ans=ans+(2**const)
                const+=1
            n=n>>1
        return ans
    
s=Solution()

n=5

print(f"{n}'s complement: {s.bitwiseComplement(n)}")

File: LeetCode/Java/ComplementOfBase10Integer.java
public class ComplementOfBase10Integer {
    public int bitwiseComplement(int n) {
        if(n == 0) return 1;
        int res = 0;
        int fac = 1;
        
        while(n != 0){
            res += fac * (n % 2 == 0 ? 1 : 0);
            fac *= 2;
            n /= 2;
        }
        return res;
    }

    public static void main(String[] args) {
        ComplementOfBase10Integer obj= new ComplementOfBase10Integer();

        int n=5;
        System.out.printf(n+"'s Complement: "+obj.bitwiseComplement(n));
    }
}
